src/main/java/it/pagopa/pn/mandate/middleware/db/MandateDao.java

```diff
@@ -478,7 +517,7 @@ private void auditLog(MandateEntity mandate, String event)
     }
 
 
-    private CompletableFuture<MandateEntity> saveHistoryAndDeleteFromMain(MandateEntity mandate, StatusEnum newstatus)
+    private CompletableFuture<MandateEntity> saveHistoryAndDeleteFromMain(MandateEntity mandate, StatusEnum newstatus, PnAuditLogEvent logEvent)
```

Comment:
questo metodo è usato per salvare "più eventi" molto simili tra loro.
puoi aggiungere direttamente qui dentro i vari logEvent in base al "newStatus" richiesto mettendo a riga 523 diciamo la creazione del log (quindi non serve farselo passare nel costruttore, usa il builder).
e poi aggiungendo a riga 533 tra transactWriteItems e thenapply, un "exceptionally" con dentro il catch - log -rethrow dell'errore